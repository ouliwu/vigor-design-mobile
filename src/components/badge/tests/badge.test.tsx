import * as React from 'react'
import { render, testA11y } from 'testing'
import Badge from '../'

describe('Badge', () => {
  test('passes a11y test', async () => {
    await testA11y(<Badge>test</Badge>)
  })

  test('render without content', () => {
    const renderer = render(<Badge content={123} data-testid='test-badge' />)
    const element = renderer.getByTestId('test-badge')
    expect(element).toHaveClass('adm-badge')
    expect(element).toMatchSnapshot()
  })

  test('render with content', () => {
    const { container } = render(<Badge content='新'>text</Badge>)
    expect(container).toMatchSnapshot()
  })

  test('renders with color', () => {
    const { container } = render(
      <Badge color='#108ee9' content='新'>
        text
      </Badge>
    )
    expect(container).toMatchSnapshot()
  })

  test('test wrapperClassName & wrapperStyle', () => {
    const { getByTestId } = render(
      <Badge
        color='#108ee9'
        content='新'
        wrapperClassName='test'
        data-testid='test-badge-wrap'
        wrapperStyle={{ color: 'red' }}
      >
        text
      </Badge>
    )
    const element = getByTestId('test-badge-wrap')
    expect(element.parentNode).toHaveClass('test')
    expect(element.parentNode).toHaveStyle('color: red')
  })
})
