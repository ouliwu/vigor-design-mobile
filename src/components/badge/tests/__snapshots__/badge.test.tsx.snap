// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge render with content 1`] = `
<div>
  <div
    class="adm-badge-wrapper"
  >
    text
    <div
      class="adm-badge adm-badge-fixed"
    >
      <div
        class="adm-badge-content"
      >
        新
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge render without content 1`] = `
<div
  class="adm-badge"
  data-testid="test-badge"
>
  <div
    class="adm-badge-content"
  >
    123
  </div>
</div>
`;

exports[`Badge renders with color 1`] = `
<div>
  <div
    class="adm-badge-wrapper"
  >
    text
    <div
      class="adm-badge adm-badge-fixed"
      style="--color: #108ee9;"
    >
      <div
        class="adm-badge-content"
      >
        新
      </div>
    </div>
  </div>
</div>
`;
